spec/features/vote_spec.rb

```diff
@@ -0,0 +1,54 @@
+require 'spec_helper'
+
+describe "Voting" do
+
+  let!(:user) { build :user }
+  let!(:post) { build :post }
+
+  before(:each) do
+    built_post = user.posts.build({ title: post.title, url: post.url })
+    built_post.save
+    login(user)
+    visit(root_url)
+  end
+
+  context "when no votes currently made" do
+
+    it "creates an upvote when a user clicks the up vote button" do
+      expect { first('.vote-button').click_on("+1") }.to change{Vote.count}.by(1)
+    end
+
+    it "creates a downvote when a user clicks the down vote button" do
+      expect { first('.vote-button').click_on("-1") }.to change{Vote.count}.by(1)
+    end
+  end
+
+  context " when a vote is currently made" do
+
+    it "it destroys the vote when an upvote is cast again" do
+      first('.vote-button').click_on("+1")
+      expect { first('.vote-button').click_on("+1") }.to change{Vote.count}.by(-1)
+    end
+
+    it "modifies the value of the existing upvote when a downvote is cast" do
+      first('.vote-button').click_on("+1")
+      vote = user.votes.first
+      first('.vote-button').click_on("-1")
+      vote = user.votes.first
```

Comment:
you're setting `vote = user.votes.first` twice here which is redundant. it might be valuable to set the vote variable after the `+1` click, and then do an expect / change assertion that clicking the `-1` changes the value of the vote record.

note: you might need to use the reload method when checking the value of your Vote object: `vote.reload.value`. Reason being, sometimes if you grab a copy of a record in a variable, and then another action changes the record, your copy might not be updated and will have outdated information. Calling `reload` refreshes it with the most up to date information.